packages/backend/src/graphql/mutations/retry-partial-step.ts

```diff
@@ -0,0 +1,45 @@
+import { ref } from 'objection'
+
+import { BadUserInputError } from '@/errors/graphql-errors'
+import ExecutionStep from '@/models/execution-step'
+import { processAction } from '@/services/action'
+
+import type { MutationResolvers } from '../__generated__/types.generated'
+
+const retryPartialStep: MutationResolvers['retryPartialStep'] = async (
+  _parent,
+  params,
+  context,
+) => {
+  const executionStep = await ExecutionStep.query()
+    .findById(params.input.executionStepId)
+    .withGraphJoined('execution')
+    .whereExists(
+      context.currentUser
+        .$relatedQuery('executions')
+        .select(1)
+        .where('executions.id', ref('execution_steps.execution_id')),
+    )
+
+  if (!executionStep) {
+    throw new BadUserInputError('Execution step not found')
+  }
+
+  if (executionStep.status !== 'success' || !executionStep.errorDetails) {
+    throw new BadUserInputError('Execution step not partially retriable')
+  }
+
+  try {
+    await processAction({
+      executionId: executionStep.executionId,
+      flowId: executionStep.execution.flowId,
+      stepId: executionStep.stepId,
+    })
```

Comment:
very nervous about running actual steps on our web tier (although tbh we do it already for test steps =.=). I think this is to mitigate the job being > 1 week old, hence it being removed from bullmq?

Main failure mode is if user is trying to send to like... 100 recipients or similar. the mutation will time out and user will be confused. We also lose record of who retried, since this technically should be a new execution step.

Would be safer to just re-construct and enqueue the job (similar to what we did for bulk retries)
```suggestion
    const {
      stepId,
      executionId,
      execution: { flowId },
      appKey,
    } = executionStep

    await enqueueActionJob({
      appKey: appKey,
      jobName: `${executionId}-${stepId}`,
      jobData: {
        flowId,
        executionId,
        stepId: stepId,
      },
      jobOptions: DEFAULT_JOB_OPTIONS,
    })
```

Reply:
Good point raised. Was thinking of that as well:

### Reasons for doing it on web server:
- It is less intensive then running test until step since this is essentially only a single step
- Single max recipients in a single action sent this year is 35. Usually the max is 23 for each month. And chances are only a minority (if any) will be blacklisted emails.
- Over the past 2 weeks, the max number of blacklisted recipients error in a day is below 20, which is an acceptable number of additional single step testing load that our web servers can handle in a day.
- moving to workers will probably require a bit more effort since we need to think of all the possible edge cases, e.g. need to prevent users from re-enqueueing if the partial retry job is already queued 
- Feedback is immediate (no 'please check back later')
### Reasons against doing on web server:
- Partial retries will respect the rate limits once we move to separate queue
- We wont need to worry about spikes in partial retries
- No risk of timeouts.

I think main thing is to get this out and monitor if spikes occur on our web server load. If so, we can easily move to enqueuing partial retries without any risks of backwards incompatibility.